dspace-server-webapp/src/main/java/org/dspace/app/rest/repository/DiscoveryRestRepository.java

```diff
@@ -103,17 +109,22 @@ public SearchResultsRest getSearchObjects(final String query, final List<String>
 
         try {
             discoverQuery = queryBuilder
-                .buildQuery(context, scopeObject, discoveryConfiguration, query, searchFilters, dsoTypes, page);
-            searchResult = searchService.search(context, scopeObject, discoverQuery);
-
-        } catch (SearchServiceException e) {
+            .buildQuery(context, scopeObject, discoveryConfiguration, query, searchFilters, dsoTypes, page);
+
+            if (dsoTypes.size() == 1 && (dsoTypes.get(0).equals(Constants.typeText[Constants.COLLECTION])
+                || dsoTypes.get(0).equals(Constants.typeText[Constants.COMMUNITY]))) {
+                searchResult = authorizeService.retrieveObjectWithAdmin(context, discoverQuery, null, null, query);
```

Comment:
Since this logic is already checking if we are searching Communities or Collections, it seems like we could potentially use `AuthorizeService.findAdminAuthorizedCollection()` and `AuthorizeService.findAdminAuthorizedCommunity()` here (or refactor those methods so that there is a version which lets us retrieve the `DiscoverResult` instead of a list of objects.

We also should add tests to prove this code is working properly. Those automated tests should be added to `DiscoveryRestRepositoryIT`.